src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/læremidler/VedtaksperiodeStatusMapper.kt

```diff
@@ -0,0 +1,46 @@
+package no.nav.tilleggsstonader.sak.vedtak.læremidler
+
+import no.nav.tilleggsstonader.sak.vedtak.læremidler.domain.Vedtaksperiode
+import no.nav.tilleggsstonader.sak.vedtak.læremidler.dto.VedtaksperiodeStatus
+
+object VedtaksperiodeStatusMapper {
+    fun settStatusPåVedtaksperioder(
+        vedtaksperioder: List<Vedtaksperiode>,
+        vedtaksperioderForrigeBehandling: List<Vedtaksperiode>?,
+    ): List<Vedtaksperiode> {
+        if (vedtaksperioderForrigeBehandling.isNullOrEmpty()) {
+            return vedtaksperioder.map { it.copy(status = VedtaksperiodeStatus.NY) }
+        }
+
+        val vedtaksperiodeIdforidForForrigeBehandling =
+            vedtaksperioderForrigeBehandling.map { it.id }
```

Comment:
Hvis denne endres til `Map<UUID, Vedtaksperiode>` så tenker jeg metodene for å sjekke om den er endret/ikke endret kan forenklest lite og ikke må bruke find på en liste for hver vedtaksperiode.

```suggestion
        val vedtaksperiodeIdforidForForrigeBehandling =
            vedtaksperioderForrigeBehandling.associateBy { it.id }
```

Med en metode på vedtaksperiode man kan då få ned sjekkene på uendrede til eks
```kotlin
val forrigeVedtaksperiode = vedtaksperiodeIdforidForForrigeBehandling[vedtaksperiode.id]
forrigeVedtaksperiode != null && !vedtaksperiode.likFomOgTom(forrigeVedtaksperiode)
```